backend/CircleForms/Controllers/UsersController.cs

```diff
@@ -1,126 +1,135 @@
+    {
+        var user = await _usersService.Get(id);
+        if (user != null)
+        {
+            return Ok(_mapper.Map<UserResponseContract>(user));
+        }
+
+        return NotFound();
+    }
+
+    /// <summary>
+    ///     Get user data.
+    /// </summary>
+    [HttpGet(ApiEndpoints.UsersGetMinimalUser)]
+    [ProducesResponseType(typeof(UserMinimalResponseContract), StatusCodes.Status200OK, "application/json")]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetMinimal([RegularExpression(@"^\d+$")] string id)
+    {
+        var user = await _cache.GetMinimalUser(id);
+        if (user != null)
+        {
+            return Ok(_mapper.Map<UserMinimalResponseContract>(user));
+        }
+
+        return NotFound();
+    }
+
+    /// <summary>
+    ///     Get all users. (Requires auth, Requires Admin role)
+    /// </summary>
+    [Authorize(Roles = "Admin")]
+    [HttpGet(ApiEndpoints.UsersGetAllUsers)]
+    public async Task<List<UserResponseContract>> GetAll()
+    {
+        _logger.LogInformation("Admin {Admin} requests users from the database", _claim);
+
+        return _mapper.Map<List<User>, List<UserResponseContract>>(await _usersService.Get());
+    }
+
+    /// <summary>
+    ///     Set user role. (Requires auth, Requires SuperAdmin role)
+    /// </summary>
+    [Authorize(Roles = "SuperAdmin")]
+    [HttpPatch(ApiEndpoints.UsersEscalateUserPrivileges)]
+    [ProducesResponseType(typeof(UserResponseContract), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    public async Task<IActionResult> EscalatePrivileges([RegularExpression(@"^\d+$")] string id, int roles)
+    {
+        var role = (Roles) roles;
+        if (role.HasFlag(Roles.SuperAdmin))
+        {
+            return BadRequest("You can not escalate user's role to SuperAdmin");
+        }
+
+        var user = await _usersService.Get(id);
+        user.Roles = role;
+
+        _logger.LogWarning("SuperAdmin {Admin} changes privileges of {Id} to {Role}", _claim,
+            id, user.Roles.ToString());
+
+        await _usersService.Update(id, user);
+
+        return Ok(_mapper.Map<UserResponseContract>(user));
+    }
+
+    /// <summary>
+    ///     Get data for current user. (Requires auth)
+    /// </summary>
+    [Authorize]
+    [HttpGet(ApiEndpoints.UsersGetMe)]
+    [ProducesResponseType(typeof(UserResponseContract), StatusCodes.Status200OK, "application/json")]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    public async Task<IActionResult> GetMe()
+    {
+        _logger.LogInformation("User {User} requests /me", _claim);
+
+        var user = await _usersService.Get(_claim);
+        if (user != null)
+        {
+            var result = _mapper.Map<UserResponseContract>(user);
+            var posts = await user.PostsRelation.ChildrenFluent().ToListAsync();

```

Comment:
Do we even need to populate posts here? Seems to be a waste of time (/me is being called quite often) and traffic. We can expose a separate endpoint for getting posts of the signed user for when we need them